mono/metadata/metadata.c

```diff
@@ -4935,8 +4934,7 @@ mono_metadata_generic_param_equal_internal (MonoGenericParam *p1, MonoGenericPar
 	 * The AOT runtime doesn't set the image when it's decoding
 	 * types, so we only compare it when the owner is NULL.
 	 */
-	if (mono_generic_param_owner (p1) == mono_generic_param_owner (p2) &&
-	    (mono_generic_param_owner (p1) || p1->image == p2->image))
+	if (mono_generic_param_owner (p1) == mono_generic_param_owner (p2))
```

Comment:
Looks like the comment above this conditional is now stale as we're no longer checking images.